crates/ra_ide_api/src/completion/completion_item.rs

```diff
@@ -44,6 +44,9 @@ pub struct CompletionItem {
     /// Additional info to show in the UI pop up.
     detail: Option<String>,
     documentation: Option<Documentation>,
+
+    /// Whether this item is marked as deprecated
+    deprecated: Option<bool>,
```

Comment:
Why is this an option? Seems like `bool` is a better fit?

Reply:
I agree. I was just trying to reflect the [specification](https://microsoft.github.io/language-server-protocol/specifications/specification-3-14/#textDocument_completion), where this field is listed as `deprecated?: boolean`, as closely as possible. But if that's not an issue, `bool` definitely sounds better.

Reply:
types inside ra_ide_api are explicitly *not* modeled after lsp. Rather, we use what makes most sense for the rust language, and explicitly lower this to lsp lowest common denominator in the conv.rs layer.

Reply:
Ah ok, sorry about that. Should be fixed now!